### PR TITLE
Indicate the existance of further trophies in special trophy list on …

### LINKED ISSUE #3273: Indicate the existance of further trophies in special trophy list on user page
… for example via a "show more trophies" button/icon/link.

see https://community.woltlab.com/thread/278296-link-zur-troph%C3%A4ensammlung-hinter-der-auflistung-der-ausgew%C3%A4hlten-troph%C3%A4en-im-ben/

### PR DESCRIPTION
**Pull request by BurntimeX**
…user page

Closes #3273

### CODE REVIEW

File: com.woltlab.wcf/templates/user.tpl [hunk 1]
```diff
@@ -197,12 +197,15 @@
...
 			<div class="contentHeaderDescription">
-				{if MODULE_TROPHY && $__wcf->session->getPermission('user.profile.trophy.canSeeTrophies') && ($user->isAccessible('canViewTrophies') || $user->userID == $__wcf->session->userID) && $user->getSpecialTrophies()|count}
+				{if MODULE_TROPHY && $__wcf->session->getPermission('user.profile.trophy.canSeeTrophies') && ($user->isAccessible('canViewTrophies') || $user->userID == $__wcf->session->userID) && $specialTrophyCount}
 					<div class="specialTrophyUserContainer">
 						<ul>
 							{foreach from=$user->getSpecialTrophies() item=trophy}
 								<li><a href="{@$trophy->getLink()}">{@$trophy->renderTrophy(32, true)}</a></li>
 							{/foreach}
+							{if $user->trophyPoints > $specialTrophyCount}
+								<li><a href="#" class="jsTooltip userTrophyOverlayList" data-user-id="{$user->userID}" title="{lang}wcf.user.trophy.showTrophies{/lang}">{lang trophyCount=$user->trophyPoints-$specialTrophyCount}wcf.user.trophy.showMoreTrophies{/lang}</a></li>
```
Comment (dtdesign) [inline com.woltlab.wcf/templates/user.tpl:207]:
```suggestion
								<li><a href="#" class="jsTooltip userTrophyOverlayList" data-user-id="{$user->userID}" title="{lang}wcf.user.trophy.showTrophies{/lang}" role="button">{lang trophyCount=$user->trophyPoints-$specialTrophyCount}wcf.user.trophy.showMoreTrophies{/lang}</a></li>
```

File: wcfsetup/install/files/lib/data/user/UserProfile.class.php [hunk 2]
```diff
@@ -431,44 +437,56 @@ public function getCurrentLocation() {
...
-			$specialTrophies = $statement->fetchAll(\PDO::FETCH_COLUMN);
+		return $this->specialTrophies;
+	}
+	
+	/**
+	 * Loads the special trophies for this user.
+	 *
+	 * @since       5.3
+	 */
+	protected function loadSpecialTrophies() {
```
Comment (dtdesign) [inline wcfsetup/install/files/lib/data/user/UserProfile.class.php:450]: This method does not serve any purpose, its content should be moved into `getSpecialTrophies()`.

File: wcfsetup/install/files/lib/data/user/UserProfile.class.php [hunk 3]
```diff
@@ -103,6 +103,12 @@ class UserProfile extends DatabaseObjectDecorator implements ITitledLinkObject {
 	 */
 	protected $coverPhoto;
 	
+	/**
+	 * @var Trophy[]
+	 * @since 5.3
+	 */
+	protected $specialTrophies;
```
Comment (joshuaruesweg) [inline wcfsetup/install/files/lib/data/user/UserProfile.class.php:110]: Saving this in a class variable causes problems when automatically marking special trophies. Currently, this is read correctly each time and if the value changes, the current value in each object is updated. Not anymore with this update. So here you should either a) continue to use the object from the UserStorageHandler (does deserialization really generate such a large overhead?) or b) add a parameter $skipCache which skips the cache (and use it in the UserTrophyAction::create).

Comment (dtdesign) [reply wcfsetup/install/files/lib/data/user/UserProfile.class.php:110]: @BurntimeX

**Pull request merged**